src/main/sf_496C0.c

```diff
@@ -1,71 +1,2995 @@
+    arg0->unk_0E8.z = gPlayer->vel.z;
+    Object_SetInfo(&arg0->info, arg0->obj.id);
+    arg0->unk_07C = 1;
+    Audio_PlaySfx(0x3100000CU, &arg0->sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+}
+
+void func_80049A9C(Object_8C* obj8c, f32 x, f32 y, f32 z) {
+    Object_8C_Initialize(obj8c);
+    obj8c->obj.status = 1;
+    obj8c->obj.id = 0x15A;
+    obj8c->timer_50 = 0x64;
+    obj8c->scale2 = 0.2f;
+    obj8c->obj.pos.x = x;
+    obj8c->obj.pos.y = y;
+    obj8c->obj.pos.z = z;
+    obj8c->unk_54.z = 40.0f;
+    obj8c->obj.rot.z = Rand_ZeroOne() * 360.0f;
+    Object_SetInfo(&obj8c->info, obj8c->obj.id);
+}
+
+void func_80049B44(void) {
+    s32 i;
+    f32 x;
+    f32 y;
+    f32 z;
+
+    for (i = 0; i < ARRAY_COUNT(gObjects8C); i++) {
+        if (!gObjects8C[i].obj.status) {
+            x = (Rand_ZeroOne() - 0.5f) * 400.0f;
+            y = (Rand_ZeroOne() - 0.5f) * 400.0f;
+            z = (-D_80177D20 - 500.0f) - Rand_ZeroOne() * 500.0f;
+            func_80049A9C(&gObjects8C[i], x, y, z);
+            break;
+        }
+    }
+}
+
+void func_80049C0C(Player* arg0) {
+    s32 var_v0;
+    char pad[0x10];
+
+    arg0->pos.x += arg0->vel.x;
+    arg0->unk_228 = 0;
+    arg0->unk_238 = 0;
+    arg0->pos.y += arg0->vel.y;
+    arg0->pos.z += arg0->vel.z;
+    func_8004990C(arg0);
+    arg0->unk_140 = -arg0->vel.z;
+    arg0->unk_144 += -arg0->vel.z;
+    D_80177D20 = arg0->unk_144;
+    Math_SmoothStepToF(&arg0->unk_130, 0.0f, 0.1f, 15.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_12C, 0.0f, 0.1f, 15.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_0EC, 0.0f, 0.1f, 5.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_0E4, 0.0f, 0.1f, 5.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_0E8, 0.0f, 0.1f, 5.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->pos.x, 0.0f, 0.1f, 50.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->pos.y, 0.0f, 0.1f, 50.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->camEye.x, 0.0f, 0.1f, 50.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->camEye.y, 0.0f, 0.1f, 50.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->camAt.x, 0.0f, 0.1f, 50.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->camAt.y, 0.0f, 0.1f, 50.0f, 0.0f);
+    Math_SmoothStepToAngle(&arg0->unk_4D8, 0.0f, 0.1f, 20.0f, 0.0f);
+    switch (arg0->unk_1D0) {
+        case 0:
+            arg0->unk_4DC = 0;
+            D_80178414 = 100.0f;
+            arg0->unk_1D0 = 1;
+            D_80178410 = 1;
+            D_80178488 = 0;
+            arg0->vel.x = 0.0f;
+            arg0->vel.y = 0.0f;
+            arg0->vel.z = -500.0f;
+            arg0->unk_0CC = -500.0f;
+            if (gTeamShields[1] > 0) {
+                func_80049968(&gObjects2F4[0], 0);
+            }
+            if (gTeamShields[2] > 0) {
+                func_80049968(&gObjects2F4[1], 1);
+            }
+            if (gTeamShields[3] > 0) {
+                func_80049968(&gObjects2F4[2], 2);
+            }
+            func_80049968(&gObjects2F4[3], 3);
+            arg0->timer_1F8 = 0x32;
+            return;
```

Comment:
same with the other ones, if it matches, also separations :)
```suggestion
break;